src/models/mail.rs

```diff
@@ -0,0 +1,85 @@
+use std::cmp::Reverse;
+
+use self::schema::mails;
+use crate::errors::{self, ZauthError};
+use crate::DbConn;
+use chrono::NaiveDateTime;
+use diesel::{self, prelude::*};
+
+use diesel::result::Error as DieselError;
+use rocket::serde::Serialize;
+use validator::Validate;
+
+pub mod schema {
+	table! {
+		use diesel::sql_types::*;
+
+		mails {
+			id -> Integer,
+			sent_on -> Timestamp,
+			subject -> Text,
+			body -> Text,
+		}
+	}
+}
+
+#[derive(Clone, Debug, Queryable, Serialize)]
+#[serde(crate = "rocket::serde")]
+pub struct Mail {
+	pub id:      i32,
+	pub sent_on: NaiveDateTime,
+	pub subject: String,
+	pub body:    String,
+}
+
+#[derive(Clone, Debug, Deserialize, FromForm, Insertable, Validate)]
+#[table_name = "mails"]
+pub struct NewMail {
+	#[validate(length(min = 3, max = 255))]
+	pub subject: String,
+	#[validate(length(min = 3))]
```

Comment:
```suggestion
	#[validate(length(min = 3, max = 10_000))]
```
Just to have a maximum, I hope if anyone wants to write an email 10 000 chars long, they would consider making a blog post out of it...